File: Week02/144_pre_order/second_run.java
 /**
尝试使用栈来解决问题
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer>result = new ArrayList<>();
        TreeNode curr = root;
        Stack<TreeNode> stack = new Stack<>();
        while(curr != null || !stack.isEmpty()) {
            if(curr==null) {
                curr = stack.pop();
            }
            result.add(curr.val);
            if(curr.right!=null){
                stack.push(curr.right);
            }
            curr = curr.left;
        }
        return result;
    }

}

File: Week02/590_post_order/second_run.cpp
/**
然后尝试用栈来解决
*/
class Solution {
public:
    vector<int> postorder(Node* root) {
        vector<int>result;
        if(root == nullptr) {
            return result;
        }
        Node* curr = root;
        stack<Node*> stack;
        stack.push(curr);
        while(!stack.empty()){
            curr = stack.top();
            stack.pop();
            result.push_back(curr->val);
            if(curr->children.size()>0) {
                for(auto node: curr->children) {
                    stack.push(node);
                }
            }
        }
        reverse(result.begin(),result.end());
        return result;
    }

};